analyzers/src/SonarAnalyzer.Common/Rules/Utilities/LogAnalyzerBase.cs

```diff
@@ -30,13 +32,19 @@ public abstract class LogAnalyzerBase<TSyntaxKind> : UtilityAnalyzerBase<TSyntax
         protected const string DiagnosticId = "S9999-log";
         private const string Title = "Log generator";
 
+        protected abstract string LanguageVersion(Compilation compilation);
```

Comment:
As there is already a GetLanguageVersion extension and it is used in multiple places, I would add a CompilationFacade to the LanguageFacade and move this retrieval there.

Reply:
This cannot be done, because they return different enums that cannot be processed in a generic way.

Reply:
Indeed I missed the fact that on the other places the enum value is used and not the string value of it.